pcsx2/DEV9/ATA/ATA_Info.cpp

```diff
@@ -32,15 +32,17 @@ void ATA::WritePaddedString(u8* data, int* index, const std::string& value, u32
 
 void ATA::CreateHDDinfo(u64 sizeSectors)
 {
-	//PS2 is limited to 32bit size HDD (2TB), however,
+	//PS2 is limited to 48bit size HDD (2TB), however,
 	//we don't yet support 48bit, so limit to 28bit size
-	constexpr u32 maxSize = (1 << 28) - 1; // 128Gb
-	sizeSectors = std::min<u32>(sizeSectors, maxSize);
+	u64 maxSize = (1 << 28) - 1; // 128Gb
+	if (lba48Supported)
+		maxSize = (1ULL << 48) - 1; // 128PiB
+
 
 	constexpr u16 sectorSize = 512;
 	DevCon.WriteLn("DEV9: HddSize : %i", sizeSectors * sectorSize / (1024 * 1024));
-	const u64 nbSectors = sizeSectors;
-	DevCon.WriteLn("DEV9: nbSectors : %i", nbSectors);
+	const u32 nbSectors = std::min<u32>(sizeSectors, maxSize); // nbSectors will hold 28-bit size
+	DevCon.WriteLn("DEV9: sizeSectors : %i", sizeSectors); // SizeSectors will keep 48-bit size
```

Comment:
You are using maxSize for limiting to the 28bit size, however, you set maxSize to the limit for LBA 48 when `lba48Supported` is true

Reply:
Fixed it